kubernetes/src/main/java/io/kubernetes/client/custom/BaseExponent.java

```diff
@@ -0,0 +1,56 @@
+package io.kubernetes.client.custom;
+
+public class BaseExponent {
+
+    private final int base;
+
+    private final int exponent;
+
+    private final Quantity.Format format;
+    public BaseExponent(int base, int exponent, Quantity.Format format) {
+        this.base = base;
+        this.exponent = exponent;
+        this.format = format;
+    }
+
+    public int getBase() {
+        return base;
+    }
+
+    public int getExponent() {
+        return exponent;
+    }
+
+    public Quantity.Format getFormat() {
+        return format;
+    }
+
+    @Override
+    public String toString() {
+        return "BaseExponent{" +
+                "base=" + base +
+                ", exponent=" + exponent +
+                ", format=" + format +
+                '}';
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        BaseExponent that = (BaseExponent) o;
+
+        if (base != that.base) return false;
+        if (exponent != that.exponent) return false;
+        return format == that.format;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = base;
```

Comment:
This is funky, and I'm not entirely sure that it is correct.  (or at least I don't understand why it is correct...

Why not:
```java
this.toString().hashCode();
```

Reply:
I must confess that this was entirely auto-generated by IntelliJ, however, I don't see any error in it.

Given our use case, I think we can happily use `this.toString().hashCode()` as the cost is not prohibitive to us and it will be more future proof in the case of adding or removing fields.